src/blueapi/service/main.py

```diff
@@ -108,6 +136,13 @@ async def on_key_error_404(_: Request, __: Exception):
     )
 
 
+async def on_token_error_401(_: Request, __: Exception):
+    return JSONResponse(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        content={},
```

Comment:
The spec wants this to return a `WWW-Authenticate` header (see [docs](https://datatracker.ietf.org/doc/html/rfc6750#section-3)). Something like `WWW-Authenticate=Bearer realm="blueapi"`? I'm not sure what the realm/scopes should be

Reply:
This specification outlines the expected response from Keycloak when it encounters an invalid token, Not for what a fast-api/any other server how it should handle invalid requests

Reply:
Pretty sure it refers to services like this

>If the protected resource request does not include authentication
   credentials or does not contain an access token that enables access
   to the protected resource, the resource server MUST include the HTTP
   "WWW-Authenticate" response header field

Or from the [HTTP status code spec](https://www.rfc-editor.org/rfc/rfc9110#status.401)

> The 401 (Unauthorized) status code indicates that the request has not been applied because it lacks valid authentication credentials for the target resource. The server generating a 401 response MUST send a [WWW-Authenticate](https://www.rfc-editor.org/rfc/rfc9110#field.www-authenticate) header field

Reply:
> [All challenges defined by this specification MUST use the auth-scheme
   value "Bearer".  This scheme MUST be followed by one or more
   auth-param values.  The auth-param attributes used or defined by this
   specification are as follows.  Other auth-param attributes MAY be
   used as well.](https://www.rfc-editor.org/rfc/rfc6750#section-3)

>  If the protected resource request included an access token and failed
   authentication, the resource server SHOULD include the "error"
   attribute to provide the client with the reason why the access
   request was declined. 


> [3.1](https://www.rfc-editor.org/rfc/rfc6750#section-3.1).  Error Codes

     When a request fails, the resource server responds using the
     appropriate HTTP status code (typically, 400, 401, 403, or 405) and
     includes one of the following error codes in the response:
  
     invalid_request
           The request is missing a required parameter, includes an
           unsupported parameter or parameter value, repeats the same
           parameter, uses more than one method for including an access
           token, or is otherwise malformed.  The resource server SHOULD
           respond with the HTTP 400 (Bad Request) status code.
  
     invalid_token
           The access token provided is expired, revoked, malformed, or
           invalid for other reasons.  The resource SHOULD respond with
           the HTTP 401 (Unauthorized) status code.  The client MAY
           request a new access token and retry the protected resource
           request.
  
     insufficient_scope
           The request requires higher privileges than provided by the
           access token.  The resource server SHOULD respond with the HTTP
           403 (Forbidden) status code and MAY include the "scope"
           attribute with the scope necessary to access the protected
           resource.
  
     If the request lacks any authentication information (e.g., the client
     was unaware that authentication is necessary or attempted using an
     unsupported authentication method), the resource server SHOULD NOT
     include an error code or other error information.



**Quotes end**:
return JsonResponse(status_code=status.HTTP_401_UNAUTHORIZED, headers={"WWW-Authenticate": "Bearer realm=blueapi"})```

seems a reasonable first attempt. According to the spec we SHOULD check what type of JWTException we get and set the `error` appropriately too.